airflow/models.py

```diff
@@ -3859,8 +3901,13 @@ def update_state(self, session=None):
         # small speed up
         if unfinished_tasks and none_depends_on_past:
             # todo: this can actually get pretty slow: one task costs between 0.01-015s
-            no_dependencies_met = all(not t.are_dependencies_met(session=session)
-                                      for t in unfinished_tasks)
+            no_dependencies_met = all(
```

Comment:
@aoen this seems to be related to the issue of "retry_delay" not being respected (https://issues.apache.org/jira/browse/AIRFLOW-747). Is this change here part of this PR or are you addressing something separately?